components/datetime/src/datetime.rs

```diff
@@ -37,19 +37,23 @@ use crate::{date::DateTimeInput, CldrCalendar, DateTimeFormatError, FormattedDat
 ///
 /// ```
 /// use icu::calendar::{DateTime, Gregorian};
-/// use icu::datetime::{TimeFormat, options::length::Time};
+/// use icu::datetime::{options::length::Time, TimeFormat};
 /// use icu::locid::locale;
-/// use icu_provider::inv::InvariantDataProvider;
 ///
-/// let provider = InvariantDataProvider;
+/// let provider = icu_testdata::get_provider();
 ///
-/// let tf = TimeFormat::<Gregorian>::try_new(locale!("en"), &provider, Time::Full, None)
-///     .expect("Failed to create DateTimeFormat instance.");
+/// let tf = TimeFormat::<Gregorian>::try_new(
+///     locale!("en"),
+///     &provider,
+///     Time::Short,
+///     None,
+/// )
+/// .expect("Failed to create DateTimeFormat instance.");
 ///
 /// let datetime = DateTime::new_gregorian_datetime(2020, 9, 1, 12, 34, 28)
 ///     .expect("Failed to construct DateTime.");
 ///
-/// let value = tf.format_to_string(&datetime);
+/// println!("{}", tf.format_to_string(&datetime));
```

Comment:
Request, here and elsewhere: Since you are in this file, could you take this opportunity to improve this doctest by putting the actual expected string in here? It is stable because the tests are based on a specific version of testdata.

Reply:
Are they? I don't want to start fixing doc tests when we bump the test data CLDR version

Reply:
One reason we have testdata is so that we can do this. We have the expected strings in most other docs tests, as you can see by browsing https://unicode-org.github.io/icu4x-docs/doc/icu/index.html. It means that the person updating the CLDR version used in testdata needs to update the strings in docs tests, but they already have to update a lot of other things including non-docs tests.

Reply:
I'd like one of the component owners to weigh in on this. Presumably expectations were consciously skipped.

Reply:
I agree with Shane, let's put an assert here.